maintenance/projects/js-toolkit/packages/liferay-npm-bundler/src/jar/index.js

```diff
@@ -29,14 +29,22 @@ export default function createJar() {
 	addSystemConfigurationFiles(zip);
 	addPortletInstanceConfigurationFile(zip);
 
-	return zip.generateAsync({type: 'nodebuffer'}).then((buffer) => {
-		fs.mkdirpSync(project.jar.outputDir.asNative);
+	const {compression, compressionOptions} = getCompressionConfiguration();
 
-		fs.writeFileSync(
-			project.jar.outputDir.join(project.jar.outputFilename).asNative,
-			buffer
-		);
-	});
+	return zip
+		.generateAsync({
+			compression,
+			compressionOptions,
```

Comment:
So this will sometimes be equivalent to `compressionOptions: undefined`. Would it be better to just omit the property in that case? ie.

```js
return zip
    .generateAsync({
        ...getCompressionConfiguration(),
        type: 'nodebuffer',
    })
```

Reply:
I sometimes wonder what goes on inside my brain... :roll_eyes: